src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -345,9 +348,39 @@ class ReferralService(
     }
   }
 
-  fun updateDraftReferral(referral: Referral, update: DraftReferralDTO): Referral {
-    validateDraftReferralUpdate(referral, update)
+  private fun updateContainsReferralDetails(update: DraftReferralDTO): Boolean {
+    return update.completionDeadline != null || update.furtherInformation != null || update.maximumEnforceableDays != null
+  }
+
+  private fun updateReferralDetails(referral: Referral, update:DraftReferralDTO, actor: AuthUser, reason: String) {
+    if (!updateContainsReferralDetails(update)) {
+      return
+    }
+
+    val existingDetails = referralDetailsRepository.findLatestByReferralId(referral.id)
+
+    val newDetails = ReferralDetails(
+      UUID.randomUUID(),
+      null,
+      referral.id,
+      OffsetDateTime.now(),
+      actor.id,
+      reason,
+      update.completionDeadline ?: existingDetails?.completionDeadline,
+      update.furtherInformation ?: existingDetails?.furtherInformation,
+      update.maximumEnforceableDays ?: existingDetails?.maximumEnforceableDays,
+    )
 
+    referralDetailsRepository.saveAndFlush(newDetails)
+
+    existingDetails?.let {
+      it.supersededBy = newDetails
+      referralDetailsRepository.saveAndFlush(it)
+    }
+
+    // TODO: currently we are duplicating these fields in both the referral, and
```

Comment:
_nit, not a blocker_
Perhaps we could move these to a function that is `@Deprecated` so we can be warned on build?

Reply:
👍